skcosmo/utils/_pcovr_utils.py

```diff
@@ -1,7 +1,62 @@
+from copy import deepcopy
+
 import numpy as np
-from scipy import linalg
+from sklearn.base import clone
+from sklearn.exceptions import NotFittedError
 from sklearn.metrics.pairwise import pairwise_kernels
 from sklearn.utils.extmath import randomized_svd
+from sklearn.utils.validation import check_is_fitted
+
+
+def check_lr_fit(regressor, X, y=None):
+    r"""
+    Checks that an regressor is fitted, and if not,
+    fits it with the provided data
+
+    :param regressor: sklearn-style regressor
+    :type regressor: object
+    :param X: feature matrix with which to fit the regressor
+        if it is not already fitted
+    :type X: array
+    :param y: target values with which to fit the regressor
+        if it is not already fitted
+    :type y: array
+    :param sample_weight: sample weights with which to fit
+        the regressor if not already fitted
+    :type sample_weight: array of shape (n_samples,)
+    """
+    try:
+        check_is_fitted(regressor)
+        fitted_regressor = deepcopy(regressor)
+
+    except NotFittedError:
+        fitted_regressor = clone(regressor)
+        fitted_regressor.fit(X, y=y)
+
+    if fitted_regressor.coef_.ndim != y.ndim:
+        raise ValueError(
+            "The target regressor has a shape incompatible "
+            "with the supplied target space"
+        )
+    elif fitted_regressor.coef_.ndim == 1:
+        if fitted_regressor.coef_.shape[0] != X.shape[1]:
+            raise ValueError(
+                "The target regressor has a shape incompatible "
+                "with the supplied feature space"
+            )
+    else:
+        if fitted_regressor.coef_.shape[0] != y.shape[1]:
+            raise ValueError(
+                "The target regressor has a shape incompatible "
+                "with the supplied target space"
+            )
+        elif fitted_regressor.coef_.shape[1] != X.shape[1]:
+            raise ValueError(
+                "The target regressor has a shape incompatible "
+                "with the supplied feature space"
+            )
```

Comment:
This can go in the initial `if` statement above.

Reply:
what `if` statement are you referring to? L36 is the first `if` statement in the file.

Reply:
ah! sorry! I meant the try statement. It's only in the case that the regressor is already fitted that we need to check all the dimensions, so this code should go within that try statement.

Reply:
I moved the chunk of code inside the `try` statement.